man/systemd.network.xml

```diff
@@ -228,6 +228,39 @@
           if <literal>RequiredForOnline=no</literal>.</para>
         </listitem>
       </varlistentry>
+      <varlistentry>
+        <term><varname>ActivationPolicy=</varname></term>
+        <listitem>
+          <para>Specifies the policy for <command>systemd-networkd</command> managing the link
+          administrative state. Specifically, this controls how <command>systemd-networkd</command>
+          changes the network device's <literal>IFF_UP</literal> flag, which is sometimes
+          controlled by system administrators by running e.g., <command>ip set dev eth0 up</command>
+          or <command>ip set dev eth0 down</command>.</para>
+
+          <para>Takes one of <literal>start-up</literal>, <literal>always-up</literal>,
+          <literal>manual</literal>, <literal>always-down</literal>, <literal>start-down</literal>,
+          or <literal>bound</literal>. When <literal>manual</literal>, <command>systemd-networkd</command>
+          will not change the link's admin state at all; the system administrator must bring the
+          interface up or down manually, as desired. When <literal>start-up</literal> or
+          <literal>start-down</literal>, <command>systemd-networkd</command> will set the link
+          admin state up or down, respectively, when <command>systemd-networkd</command> starts
+          (or, when the interface is first detected, if that occurs after <command>systemd-networkd</command>
+          starts), but it will not change the flag after the first time. When <literal>always-up</literal>
```

Comment:
Hmm, networkd now supports configuration reloads, so it may bring interfaces up not only at start. So instead of referring to networkd startup, say "... will set the administrative state of the link to up or down after configuring the interface, but will not change the state afterwards".

Also, with "start-up", if the interface is manually set to down, will `networkctl reconfigure` bring it back up? I think it should.

Reply:
I updated the doc, to include clarifying it 'up' and 'down' will apply only during (re)configuration; let me know if you think the new wording should be changed.

> Also, with "start-up", if the interface is manually set to down, will networkctl reconfigure bring it back up? I think it should.

I agree; I moved the flag out of the link and into the link->network, so up/down policy should be applied each reconfiguration.

Reply:
moving the ->activated flag into link->network was a mistake on my part (since ->network can be shared between links and shouldn't hold link state, i'm pretty sure this was causing the test failures), but i just moved it back to link->activated and it should still be reset on networkctl reload, since that will call link_reconfigure() with force=True.